spec/integration/status_updates_spec.rb

```diff
@@ -1,69 +1,54 @@
 RSpec.describe "Receiving a status update", type: :request do
-  let(:delivery_attempt) { create(:delivery_attempt, status: "sent") }
-  let(:reference) { delivery_attempt.id }
-
-  let(:permissions) { %w[signin status_updates] }
-  let(:user) { create(:user, permissions: permissions) }
-  before { login_as(user) }
-
   describe "#create" do
+    let(:status) { "delivered" }
+    let(:permissions) { %w[signin status_updates] }
+    let(:user) { create(:user, permissions: permissions) }
     let(:params) do
       {
-        reference: reference,
-        status: "delivered",
+        reference: SecureRandom.uuid,
+        status: status,
+        to: "test.user@example.com",
       }
     end
 
-    let(:permissions) { %w[signin status_updates] }
-
-    it "calls the status update service" do
-      expect(StatusUpdateService).to receive(:call).with(
-        reference: reference,
-        status: "delivered",
-        user: user,
-      )
-
-      post "/status-updates", params: params
-    end
+    before { login_as(user) }
 
-    it "renders 204 no content" do
+    it "responds with no content" do
       post "/status-updates", params: params
 
-      expect(response.status).to eq(204)
+      expect(response).to have_http_status(:no_content)
       expect(response.body).to eq("")
     end
 
-    it "updates the delivery attempt" do
-      expect { post "/status-updates", params: params }
-        .to change { delivery_attempt.reload.status }
-        .to eq("delivered")
-    end
-
     context "when a user does not have 'status_updates' permission" do
       let(:permissions) { %w[signin] }
 
-      it "renders 403" do
+      it "responds with a forbidden" do
         post "/status-updates", params: params
 
-        expect(response.status).to eq(403)
+        expect(response).to have_http_status(:forbidden)
       end
     end
 
-    context "when a status attempt arrives for an already handled delivery attempt" do
-      before { delivery_attempt.update!(status: "delivered") }
+    context "when a status attempt arrives with an permanent-failure status" do
+      let(:status) { "permanent-failure" }
 
-      it "renders 409" do
-        post "/status-updates", params: params
+      it "can unsubscribe all of a subscribers subscriptions by email id" do
```

Comment:
We're missing a test for when the subscriber doesn't exist. (Looks like the previous class was missing it also, so I can understand why.)

Reply:
Thanks - it was one of those rather annoying negative tests, I'm always in two minds about adding tests that test code doesn't do something.

Reply:
Well, at least it's in good spirit: we could have code that would error otherwise.